crates/matrix-sdk/src/sliding_sync/list/mod.rs

```diff
@@ -369,10 +372,19 @@ impl SlidingSyncListInner {
     /// state of the request generator.
     #[instrument(skip(self), fields(name = self.name, ranges = ?&self.ranges))]
     fn request(&self) -> v4::SyncRequestList {
-        let ranges = self.request_generator.read().unwrap().ranges.clone();
+        use ruma::UInt;
+        let ranges = self
+            .request_generator
+            .read()
+            .unwrap()
+            .ranges
+            .iter()
+            .map(|r| (UInt::new(*r.start()).unwrap(), UInt::new(*r.end()).unwrap()))
```

Comment:
Don't we want to use `u32` for the ranges as well to get rid of these unwraps? Do we need an `u64`?